### PR TITLE
[WIP] Fix issues related to displaying the correct school name

### PR DESCRIPTION
**Pull request by clareconstantine**
- List View should show the school name and district name from the application, not the applicant's account

### CODE REVIEW

**Comment by clareconstantine** [comment]
@aoby this was a quick fix for this issue, but involves making more db calls. Do you think it is worth storing the district and school as fields on the application? or doing something else to prevent making these calls?

File: dashboard/app/models/pd/application/application_base.rb [hunk 1]
```diff
@@ -255,5 +255,10 @@ def total_score
     def include_additional_text(hash, field_name, *options)
       hash[field_name] = answer_with_additional_text hash, field_name, *options
     end
+
+    def school
+      school_id = sanitize_form_data_hash[:school].to_i
+      School.find_by_id(school_id)
```
Comment (aoby) [inline dashboard/app/models/pd/application/application_base.rb]: Make sure to handle -1 school id, which means we had a custom school. In that case, school name is specified as text and district will be missing.

File: dashboard/app/models/pd/application/application_base.rb [hunk 2]
```diff
@@ -230,11 +230,11 @@ def regional_partner_name
     end
 
     def school_name
-      user.try(:school_info).try(:effective_school_name).try(:titleize)
+      school.try(:name).try(:titleize)
     end
 
     def district_name
-      user.try(:school_info).try(:effective_school_district_name).try(:titleize)
+      District.find_by_id(school.try(:school_district_id)).try(:name).try(:titleize)
```
Comment (aoby) [inline dashboard/app/models/pd/application/application_base.rb]: We want `SchoolDistrict`, not `District` here (District is a leftover part of the old ops dashboard - deprecated and should be removed when we have a chance).

On that note, you can just use the [school_district](https://github.com/code-dot-org/code-dot-org/blob/3cb30d8eefe6c45bb6b3031a8b60da56dc6553f5/dashboard/app/models/school.rb#L33) association: 
```ruby
school.try(:school_district).try(:name).try(:titleize)
```

**Comment by clareconstantine** [comment]
@aoby this will be for list view as well as detail view. should we change approaches, or wait and see if it is slow?

**Comment by aoby** [comment]
As discussed offline, let's go with this for now and open a lower pri task for later optimization by storing the names in the properties json (lazily).

**Pull request merged**